src/model/model.hpp

```diff
@@ -144,6 +146,13 @@ class Model {
     /// Returns the number of constraints in the model
     [[nodiscard]] size_t NumConstraints() const { return this->constraints_.size(); }
 
+    // Returns a State object initialized from the model nodes
+    State CreateState() {
```

Comment:
This should be `[[nodiscard]]`

Reply:
I don't disagree, could you explain what it means in this context? I thought this tag was so a function that isn't called wasn't discarded.

Reply:
nodiscard means that it is a compile time error to call this function and not store the returned value.  So `model.CreateState();` is an error, but `auto state = model.CreateState();` is not.  Since this function doesn't have side effects, it can be assumed that discarding the return value is a mistake by the user.  

I think this method can actually have a trailing const, too, since it doesn't modify model.

Reply:
Thanks for the explanation, I understand why you'd want the `[[nodiscard]]` here. I've added the `const` postfix.